src/test/java/org/example/action/oauth/TokenAction.java

```diff
@@ -64,6 +70,13 @@ public String post() {
       case client_secret_basic -> assertEquals(httpRequest.getHeader("Authorization"), "Basic dGhlIGNsaWVudCBJRDp0aGUgY2xpZW50IHNlY3JldA==");
       case none -> assertTrue(true);
     }
+
+    // if additional parameters were sent, validate them
+    MockOAuthUserLoginSecurityContext.additionalParameters.forEach((key, value) -> {
+      assertTrue(unknownParameters.containsKey(key), "Missing additional parameter: " + key);
+      assertEquals(unknownParameters.get(key), value.toArray(), "Mismatched additional parameter value for: " + key);
+    });
```

Comment:
To be somewhat consistent with similar tests, I think you'd want to make the `unknownParameters` parameters map static so you can assert on it in your test instead of asserting here. 

See `SecureAction.UnknownParameters` as an example, and this is then reset in `PrimeBaseTest.beforeMethod` by calling `SecureAction.UnknownParameters.clear();`